var/spack/repos/builtin/packages/phist/package.py

```diff
@@ -175,7 +173,8 @@ class Phist(CMakePackage):
     # Python 3 or later is required for generating the Fortran 2003 bindings
     # since version 1.7, you can get rid of the dependency by switching off
     # the feature (e.g. use the '~fortran' variant)
-    depends_on("python@3:", when="@1.7: +fortran", type="build")
+    # For the upperbound see https://bitbucket.org/essex/phist/issues/246/does-not-build-with-python-311
+    depends_on("python@3:3.10", when="@1.7: +fortran", type="build")
```

Comment:
Do we have a preference for depends_on vs. conflicts? It used to be the case that depends_on guided the concretizer and conflicts crashed it, but that's no longer the case for clingo. I usually use depends_on for what is documented as a dep and conflicts for surprises we find (aka bugs).